### PR TITLE
feat: upgrade goldentest to v1beta4

### PR DESCRIPTION
**Pull request by capri-xiyue**
fixed part of #428 
1. Upgraded goldentest to v1beta4
2. Added UT to guard unreleased version.

### CODE REVIEW

File: templates/commands/goldentest/test_funcs_test.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/google/go-cmp/cmp/cmpopts"
 
 	"github.com/abcxyz/abc/templates/model"
-	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1beta3"
+	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1beta4"
```
Comment (sailorlqh) [inline templates/commands/goldentest/test_funcs_test.go:27]: is this `v1beta4` expected?

Comment (capri-xiyue) [reply templates/commands/goldentest/test_funcs_test.go:27]: This is expected

File: templates/model/goldentest/v1beta3/upgrade.go [hunk 2]
```diff
@@ -16,15 +16,21 @@ package goldentest
...
+	v1beta4 "github.com/abcxyz/abc/templates/model/goldentest/v1beta4"
 )
 
 // Upgrade implements model.ValidatorUpgrader.
 func (t *Test) Upgrade(ctx context.Context) (model.ValidatorUpgrader, error) {
-	logger := logging.FromContext(ctx).With("logger", "Upgrade")
-	logger.DebugContext(ctx, "finished upgrading goldentest model, this is the most recent version")
+	var out v1beta4.Test
+
+	if err := copier.Copy(&out, t); err != nil {
```
Comment (drevell) [inline templates/model/goldentest/v1beta3/upgrade.go:31]: I think we need to do something like `SkipStdout` here to make sure that a new binary doesn't attempt to verify the stdout of old tests. Though probably you have this planned for a separate PR.

**Review by drevell — APPROVED** [review]

File: templates/model/goldentest/v1beta3/upgrade.go (hunk 2 above)
Comment (capri-xiyue) [reply templates/model/goldentest/v1beta3/upgrade.go:31]: I will do it in another PR. This PR focuses on gitkeep ut and goldentest upgrade

**Pull request merged**